scripts/tools/SetupUserSynchronizer.php

```diff
@@ -0,0 +1,102 @@
+use oat\taoEncryption\Service\Sync\EncryptUserSyncFormatter;
+use oat\taoSync\model\synchronizer\user\administrator\RdfAdministratorSynchronizer;
+use oat\taoSync\model\synchronizer\user\proctor\RdfProctorSynchronizer;
+use oat\taoSync\model\synchronizer\user\testtaker\RdfTestTakerSynchronizer;
+use oat\taoSync\model\synchronizer\user\UserSynchronizer;
+use oat\taoSync\model\SyncService;
+use common_report_Report as Report;
+
+/**
+ * Class SetupAsymmetricKeys
+ * @package oat\taoEncryption\tools
+ *
+ * sudo -u www-data php index.php 'oat\taoEncryption\scripts\tools\SetupUserSynchronizer'
+ */
+class SetupUserSynchronizer extends InstallAction
+{
+    /**
+     * @param $params
+     * @return Report
+     * @throws \common_Exception
+     */
+    public function __invoke($params)
+    {
+        /** @var \common_ext_ExtensionsManager $extensionManager */
+        $extensionManager = $this->getServiceLocator()->get(\common_ext_ExtensionsManager::SERVICE_ID);
+
+        if (!$extensionManager->isInstalled('taoSync')) {
+            return Report::createSuccess('Cannot setup sync, to taoSync extension installed');
+        }
+
+        $encryptUserSyncFormatter = new EncryptUserSyncFormatter([
+            EncryptUserSyncFormatter::OPTION_ENCRYPTION_SERVICE => EncryptionSymmetricService::SERVICE_ID,
+            EncryptUserSyncFormatter::OPTION_ENCRYPTION_KEY_PROVIDER_SERVICE => SimpleKeyProviderService::SERVICE_ID,
+            EncryptUserSyncFormatter::OPTION_ENCRYPTED_PROPERTIES => [
+                OntologyRdfs::RDFS_LABEL,
+                \oat\generis\model\GenerisRdf::PROPERTY_USER_FIRSTNAME,
+                \oat\generis\model\GenerisRdf::PROPERTY_USER_LASTNAME,
+                \oat\generis\model\GenerisRdf::PROPERTY_USER_MAIL
+            ],
+        ]);
+
+        $this->registerService(EncryptUserSyncFormatter::SERVICE_ID, $encryptUserSyncFormatter);
+
+        /** @var SyncService $syncService */
+        $syncService = $this->getServiceLocator()->get(SyncService::SERVICE_ID);
+        $synchronizers = $syncService->getOption(SyncService::OPTION_SYNCHRONIZERS);
+        $syncOptions = $syncService->getOptions();
+
+        /** @var UserSynchronizer $administrator */
+        $administrator = $syncOptions[SyncService::OPTION_SYNCHRONIZERS]['administrator'];
+        $administrator->setOption(UserSynchronizer::OPTIONS_FORMATTER_CLASS , EncryptUserSyncFormatter::SERVICE_ID);
+
+        /** @var UserSynchronizer $proctor */
+        $proctor = $syncOptions[SyncService::OPTION_SYNCHRONIZERS]['proctor'];
+        $proctor->setOption(UserSynchronizer::OPTIONS_FORMATTER_CLASS, EncryptUserSyncFormatter::SERVICE_ID);
+
+        /** @var UserSynchronizer $testTaker */
+        $testTaker = $syncOptions[SyncService::OPTION_SYNCHRONIZERS]['test-taker'];
+        $testTaker->setOption(UserSynchronizer::OPTIONS_FORMATTER_CLASS, EncryptUserSyncFormatter::SERVICE_ID);
+
+        $syncService->setOptions(array_merge(
```

Comment:
you don't need to set options, you have already done

Reply:
no necessary